src/setup/rgw.py

```diff
@@ -251,3 +266,69 @@ def _create_rgw_user(conf):
             else:
                 raise SetupError(rc, f'"{create_usr_cmd}" failed with error {err}.')
+
+    @staticmethod
+    def _create_symbolic_link_fid(conf: MappedConf):
+        """ Create symbolic link of FID sysconfig files."""
+        base_config_path = conf.get(CONFIG_PATH_KEY)
+        sysconfig_file_path = os.path.join(base_config_path, 'rgw', 'sysconfig', Rgw._machine_id)
+        file_name = sysconfig_file_path + '/rgw-0x*'
+        list_matching = []
+        for name in glob.glob(file_name):
+            list_matching.append(name)
+        count = len(list_matching)
+        Log.info(f'rgw FID file count : {count}')
+        if count < 1:
+           raise Exception(f'HARE-sysconfig file is missing at {sysconfig_file_path}')
+
+        # Create symbolic links of rgw-fid files created by hare.
+        # e.g rgw-0x7200000000000001\:0x9c -> rgw-1 , rgw-0x7200000000000001\:0x5b -> rgw-2
+        index = 1
+        for src_path in list_matching:
+            file_name = 'rgw-' + str(index)
+            dst_path = os.path.join(sysconfig_file_path, file_name)
+            Rgw._create_symbolic_link(src_path, dst_path)
+            index += 1
+
+    @staticmethod
+    def _create_symbolic_link(src_path: str, dst_path: str):
+        """create symbolic link."""
+        Log.info(f'symbolic link source path: {src_path}')
+        Log.info(f'symbolic link destination path: {dst_path}')
+        if os.path.exists(dst_path):
+           Log.info('symbolic link is already present')
+           os.unlink(dst_path)
+           Log.info('symbolic link is unlinked')
+        os.symlink(src_path, dst_path)
+        Log.info(f'symbolic link created successfully')
+
+    @staticmethod
+    def _parse_endpoint_values(conf, rgw_instance_name: str):
+        """Read sysconfig file generated by hare
+         1) Read symblink file '{rgw_instance_name}' as default endpoints in config phase.
+         2) fetch endpoint values for running radosgw-admin tool.
+        """
+        base_config_path = conf.get(CONFIG_PATH_KEY)
+        sysconfig_file_path = os.path.join(base_config_path, 'rgw', 'sysconfig', Rgw._machine_id)
+        endpoint_file = os.path.join(sysconfig_file_path, rgw_instance_name)
+        endpoints = {}
+        with open(endpoint_file) as ep_file:
+            for line in ep_file:
+                ep_name, ep_value = line.partition('=')[::2]
+                endpoints[ep_name.strip()] = str(ep_value.strip())
+
+        return endpoints
+
+    @staticmethod
+    def _update_rgw_config_with_endpoints(conf, endpoints: dict):
+        """Update endpoint values to rgw config file."""
+        rgw_config_dir = Rgw._get_rgw_config_dir(conf)
+        rgw_config_file = os.path.join(rgw_config_dir, RGW_CONF_FILE)
+        Rgw._load_rgw_config(Rgw._rgw_conf_idx, f'ini://{rgw_config_file}')
+        Conf.set(Rgw._rgw_conf_idx, 'client>motr profile fid', endpoints['MOTR_PROFILE_FID'])
```

Comment:
if any one of the expected endpoint kv pair is missing, we should fail out. Currently its without any validation

Reply:
we have added a validation code for checking the config parameter is present or not  and its value should not be empty